Java/core/src/main/java/com/amazon/randomcutforest/IStateMapper.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+public interface IStateMapper<Model, State> {
+    State toState(Model model);
+
+    default Model toModel(State state) {
+        return toModel(state, 0L);
+    }
+
+    default Model toModel(State state, long seed) {
+        return toModel(state);
+    }
```

Comment:
Question: Since java supports function overloading, having #to() functions makes the code more readable:

```
public interface IStateMapper<Model, State> {
    State to(Model model);

    default Model from(State state) {
        return toModel(state, 0L);
    }

    default Model from(State state, long seed) {
        return toModel(state);
    }
}
```

Reply:
The `to` method doesn't read very well to me. I think it's because the object of "to" is CompactRandomCutTreeState, which is all the way to the left. If you read this line in your head, you say "to tree", which is the opposite of what we're doing.

```
CompactRandomCutTreeState state = mapper.to(tree);
```

The `from` method reads well, though. Would it be weird to use `from` in both directions?

```
CompactRandomCutTreeState state = mapper.from(tree);
CompactRandomCutTree tree = mapper.from(state);
```

Reply:
Speaking for me, a "from" in both directions make sense -- but what if there are more than one target? It's hypothetical though, in the context of this library. Why not use mapper.toRandomCutTreeState ?

Reply:
These methods are defined by an interface with generic type parameters, so the method names are also "generic". Hence `toState` and `toModel`.